src/video.rs

```diff
@@ -115,34 +115,53 @@ fn pretty_spinner_style() -> ProgressStyle {
         .progress_chars(PROGRESS_CHARS)
 }
 
+type VideoCompareMutex<E, F> = Arc<Mutex<VideoCompare<E, F>>>;
+
+struct VideoCompare<E: Decoder, F: Decoder> {
+    current_frame: usize,
+    next_frame: usize,
+    source: E,
+    distorted: F,
+}
+
 fn calc_score<S: Pixel, D: Pixel, E: Decoder, F: Decoder>(
-    mtx: &Mutex<(usize, (E, F))>,
+    mtx: &VideoCompareMutex<E, F>,
     src_yuvcfg: &YuvConfig,
     dst_yuvcfg: &YuvConfig,
     inc: usize,
+    end_frame: Option<usize>,
     verbose: bool,
 ) -> Option<(usize, f64)> {
     let (frame_idx, (src_frame, dst_frame)) = {
         let mut guard = mtx.lock().unwrap();
-        let curr_frame = guard.0;
 
-        let src_frame = guard.1 .0.read_video_frame::<S>();
-        let dst_frame = guard.1 .1.read_video_frame::<D>();
+        let skip_frames = guard.next_frame - guard.current_frame;
```

Comment:
The logic is correct (when iterating over `1..skip_frames`), but the variable name isn't great. E.g. if `current_frame` is 10 and `next_frame` is 11, `skip_frames` should logically be 0 (we do not need to skip any frames). But it's 1 with this code.

Can you rename the variable to something like `frames_until_next_frame` or maybe `distance_to_next`? Would make it easier to understand I think

Reply:
Renamed it to `distance_to_next`